File: src/main/audio-formats.ts
export const AUDIO_EXTENSIONS = new Set([
  ".mp3",
  ".flac",
  ".wav",
  ".ogg",
  ".oga",
  ".aac",
  ".m4a",
  ".wma",
  ".opus",
  ".webm",
  ".aiff",
  ".aif",
  ".mka",
  ".mp2",
]);

export const MIME_TYPES: Record<string, string> = {
  mp3: "audio/mpeg",
  mp2: "audio/mpeg",
  m4a: 'audio/mp4; codecs="mp4a.40.2"',
  mp4: 'audio/mp4; codecs="mp4a.40.2"',
  aac: "audio/aac",
  ogg: "audio/ogg",
  oga: "audio/ogg",
  opus: "audio/ogg",
  wav: "audio/wav",
  flac: "audio/flac",
  webm: "audio/webm",
  mka: "audio/x-matroska",
};

// Formats Chromium plays directly via <audio> + Range. m4a/mp4 are NOT
// listed here because their playability depends on whether the moov atom
// precedes mdat (faststart) — that runtime check lives in
// transcode.ts:shouldTranscode.
export const NATIVE_FORMATS = new Set([
  "mp3",
  "wav",
  "flac",
  "ogg",
  "oga",
  "opus",
  "aac",
  "webm",
]);

File: src-tauri/src/audio_formats.rs
#![allow(dead_code)]

pub const AUDIO_EXTENSIONS: &[&str] = &[
    "mp3", "flac", "wav", "ogg", "oga", "aac", "m4a", "wma", "opus", "webm", "aiff", "aif", "mka",
    "mp2",
];

pub fn is_audio_extension(ext: &str) -> bool {
    let lower = ext.to_ascii_lowercase();
    AUDIO_EXTENSIONS.iter().any(|e| *e == lower)
}

pub fn mime_for(format: &str) -> &'static str {
    match format.to_ascii_lowercase().as_str() {
        "mp3" | "mp2" => "audio/mpeg",
        "m4a" | "mp4" => "audio/mp4; codecs=\"mp4a.40.2\"",
        "aac" => "audio/aac",
        "ogg" | "oga" | "opus" => "audio/ogg",
        "wav" => "audio/wav",
        "flac" => "audio/flac",
        "webm" => "audio/webm",
        "mka" => "audio/x-matroska",
        _ => "application/octet-stream",
    }
}

pub fn is_native_format(format: &str) -> bool {
    matches!(
        format.to_ascii_lowercase().as_str(),
        "mp3" | "wav" | "flac" | "ogg" | "oga" | "opus" | "aac" | "webm"
    )
}
